src/lib/redux-store/timeseries/timeseries-apis.ts

```diff
@@ -1,47 +1,61 @@
-import type { ActiveValues } from "$lib/redux-store/metrics-leaderboard/metrics-leaderboard-slice";
-import type { MeasureDefinitionEntity } from "$common/data-modeler-state-service/entity-state-service/MeasureDefinitionStateService";
+import type { ActiveValues } from "$lib/redux-store/explore/explore-slice";
+import type {
+  BasicMeasureDefinition,
+  MeasureDefinitionEntity,
+} from "$common/data-modeler-state-service/entity-state-service/MeasureDefinitionStateService";
 import { createAsyncThunk } from "$lib/redux-store/redux-toolkit-wrapper";
 import { EntityType } from "$common/data-modeler-state-service/entity-state-service/EntityStateService";
 import { streamingFetchWrapper } from "$lib/util/fetchWrapper";
 import type { TimeSeriesResponse } from "$common/database-service/DatabaseTimeSeriesActions";
 import { updateTimeSeries } from "$lib/redux-store/timeseries/timeseries-slice";
-import type { MetricsLeaderboardEntity } from "$lib/redux-store/metrics-leaderboard/metrics-leaderboard-slice";
+import type { MetricsExploreEntity } from "$lib/redux-store/explore/explore-slice";
 import type { RillReduxState } from "$lib/redux-store/store-root";
 import { prune } from "../../../routes/_surfaces/workspace/explore/utils";
 import type { RollupInterval } from "$common/database-service/DatabaseColumnActions";
+import { selectMeasureById } from "$lib/redux-store/measure-definition/measure-definition-selectors";
 
 export const generateTimeSeriesApi = createAsyncThunk(
   `${EntityType.MetricsLeaderboard}/generateTimeSeries`,
   async (
     {
-      metricsDefId,
+      id,
       measures,
       filters,
       pixels,
       rollupInterval,
     }: {
-      metricsDefId: string;
-      measures: Array<MeasureDefinitionEntity>;
+      id: string;
+      measures?: Array<MeasureDefinitionEntity>;
       filters?: ActiveValues;
       pixels?: number;
       rollupInterval?: RollupInterval;
     },
     thunkAPI
   ) => {
+    const state = thunkAPI.getState() as RillReduxState;
+    const metricsLeaderboard: MetricsExploreEntity =
+      state.metricsLeaderboard.entities[id];
     if (!filters) {
-      const metricsLeaderboard: MetricsLeaderboardEntity = (
-        thunkAPI.getState() as RillReduxState
-      ).metricsLeaderboard.entities[metricsDefId];
       filters = prune(metricsLeaderboard.activeValues);
     }
+    if (!measures) {
+      measures = metricsLeaderboard.measureIds.map((measureId) =>
+        selectMeasureById(state, measureId)
+      );
+    }
+
     const stream = streamingFetchWrapper<TimeSeriesResponse>(
-      `metrics/${metricsDefId}/time-series`,
+      `metrics/${id}/time-series`,
       "POST",
       {
-        expressionEntries: measures.map((measure) => [
-          measure.id,
-          measure.expression,
-        ]),
+        measures: measures.map(
```

Comment:
why is this map needed?

Reply:
It filters out data from measure that is not needed. Depends on the deserialiser used in Java but in my experience additional fields can sometimes break it.